src/Chapter2.hs

```diff
@@ -858,9 +876,13 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
-
-
+rewind :: [a] -> [a]
+rewind [] = []
+rewind l = go [] l
+  where
+    go :: [a] -> [a] -> [a]
+    go acc [] = acc
+    go acc xs = foldl (flip (:)) acc xs
```

Comment:
Wow, that's a clever solution using standard functions! 🤯

Btw, you don't need to handle the case with `[]` in the `go` function, since `foldl` already takes care of this. And, after you do this, you can notice, that there is no need in the `go` function with such implementation at all, and you can implement `rewind` in a single line 🙂